Source/monster.cpp

```diff
@@ -2649,14 +2614,10 @@ void GargoyleAi(int monsterId)
 	assert(static_cast<size_t>(monsterId) < MaxMonsters);
 	auto &monster = Monsters[monsterId];
 
-	int dx = monster.position.tile.x - monster.position.last.x;
-	int dy = monster.position.tile.y - monster.position.last.y;
 	Direction md = GetMonsterDirection(monster);
 	if (monster.activeForTicks != 0 && (monster.flags & MFLAG_ALLOW_SPECIAL) != 0) {
 		UpdateEnemy(monster);
-		int mx = monster.position.tile.x - monster.enemyPosition.x;
-		int my = monster.position.tile.y - monster.enemyPosition.y;
-		if (abs(mx) < monster.intelligence + 2 && abs(my) < monster.intelligence + 2) {
+		if (monster.shortestDistanceToEnemy() < monster.intelligence + 2) {

```

Comment:
This should be longest, meaning there is no use of shortestDistanceToEnemy() and it can be removed.
```suggestion
		if (monster.longestDistanceToEnemy() < monster.intelligence + 2) {
```